parser.go

```diff
@@ -1072,7 +1124,16 @@ func (p *parser) getSchemaObjectCached(pkgPath, pkgName, typeName string) (*Sche
 	var schemaObject *SchemaObject
 
 	// see if we've already parsed this type
-	if knownObj, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, typeName)]; ok {
+	knownObj := checkCache(pkgName, typeName, p)
+	if knownObj != nil {
+		schemaObject = knownObj
+	}
+
+	if schemaObject != nil {
+		return schemaObject, nil
+	}
+
+	if knownObj, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, typeName, p)]; ok {
```

Comment:
Seems weird to pass `p` in as an arg when the function is being executed on `p` already